public/apps/configuration/panels/internal-user-edit/attribute-panel.tsx

```diff
@@ -0,0 +1,122 @@
+ *   permissions and limitations under the License.
+ */
+
+import {
+  EuiButton,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiFormRow,
+} from '@elastic/eui';
+import { map } from 'lodash';
+import React, { Dispatch, Fragment, SetStateAction } from 'react';
+import { UserAttributes } from '../../types';
+import {
+  appendElementToArray,
+  removeElementFromArray,
+  updateElementInArrayHandler,
+} from '../../utils/array-state-utils';
+import { PanelWithHeader } from '../../utils/panel-with-header';
+import { UserAttributeStateClass } from './types';
+import { FormRow } from '../../utils/form-row';
+
+export function buildAttributeState(attributesDict: UserAttributes): UserAttributeStateClass[] {
+  return map(attributesDict, (v, k) => ({
+    key: k || '',
+    value: v,
+  }));
+}
+
+export function unbuildAttributeState(attributesList: UserAttributeStateClass[]): UserAttributes {
+  return attributesList.reduce(
+    (attributes: UserAttributes, { key, value }: UserAttributeStateClass) => ({
+      ...attributes,
+      [key]: value,
+    }),
+    {}
+  );
+}
+
+function getEmptyAttribute() {
+  return { key: '', value: '' };
+}
+
+function generateAttributesPanels(
+  userAttributes: UserAttributeStateClass[],
+  setAttributes: Dispatch<SetStateAction<UserAttributeStateClass[]>>
+) {
+  const panels = userAttributes.map((userAttribute, arrayIndex) => {
+    const onValueChangeHandler = (attributeToUpdate: string) =>
+      updateElementInArrayHandler(userAttributes, [arrayIndex, attributeToUpdate]);
+
+    return (
+      <Fragment key={`attributes-${arrayIndex}`}>
+        <EuiFlexGroup justifyContent="flexStart">
+          <EuiFlexItem grow={false}>
+            <FormRow headerText={arrayIndex === 0 ? 'Variable name' : ''}>
+              <EuiFieldText
+                onChange={onValueChangeHandler('key')}
+                placeholder="Type in variable name"
+              />
+            </FormRow>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <FormRow headerText={arrayIndex === 0 ? 'Value' : ''}>
+              <EuiFieldText onChange={onValueChangeHandler('value')} placeholder="Type in value" />
+            </FormRow>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFormRow hasEmptyLabelSpace={arrayIndex === 0 ? true : false}>
+              <EuiButton
+                color="danger"
+                onClick={() => removeElementFromArray(setAttributes, [], arrayIndex)}
+              >
+                Remove
+              </EuiButton>
+            </EuiFormRow>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </Fragment>
+    );
+  });
+  return <>{panels}</>;
+}
+
+export function AttributePanel(props: {
+  state: UserAttributeStateClass[];
+  setState: Dispatch<SetStateAction<UserAttributeStateClass[]>>;
+}) {
+  const { state, setState } = props;
+  return (
+    <PanelWithHeader
+      headerText="Attributes - optional"
```

Comment:
From the mockup, "optional" has a different font

Reply:
Updated, didn't cancel the bold since it is controlled by the EuiTitle.